CDCgov/prime-reportstream
# Engagement/nj prework

This PR adds NJ pre-work and integration test for testing NJ receiver transforms and receiver settings.

Test Steps:
1. Run `./gradlew testIntegration`
2. Run `./prime multiple-settings set --input ./settings/STLTs/NJ/nj-doh.yml --env local`
3. Send a report routed to NJ
4. Verify report gets routed to NJ

## Changes
- Created transform for NJ
- Included settings for new UP receiver for NJ

## Checklist

### Testing
- [X] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container? (didn't change 

## Linked Issues
- Fixes #13855 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #14264 

<img width="1892" alt="image" src="https://github.com/CDCgov/prime-reportstream/assets/39102333/32a0b6fe-5625-4047-aa5e-83f0152c85f8">
